apps/hash-frontend/src/pages/shared/top-context-bar/share-dropdown-menu/invite-account-form.tsx

```diff
@@ -0,0 +1,148 @@
+import { Autocomplete, Avatar } from "@hashintel/design-system";
+import { AccountGroupId, AccountId } from "@local/hash-subgraph";
+import {
+  autocompleteClasses,
+  Box,
+  outlinedInputClasses,
+  Typography,
+} from "@mui/material";
+import { FunctionComponent, useCallback, useMemo, useState } from "react";
+
+import { useOrgs } from "../../../../components/hooks/use-orgs";
+import { useOrgsWithLinks } from "../../../../components/hooks/use-orgs-with-links";
+import { useUsers } from "../../../../components/hooks/use-users";
+import { useUsersWithLinks } from "../../../../components/hooks/use-users-with-links";
+import {
+  MinimalOrg,
+  MinimalUser,
+  Org,
+  User,
+} from "../../../../lib/user-and-org";
+import { Button } from "../../../../shared/ui";
+import { getImageUrlFromEntityProperties } from "../../get-image-url-from-properties";
+
+export const InviteAccountForm: FunctionComponent<{
+  excludeAccountIds?: (AccountId | AccountGroupId)[];
+  onInviteAccount: (account: MinimalOrg | MinimalUser) => void;
+}> = ({ onInviteAccount, excludeAccountIds }) => {
+  const [selectedAccount, setSelectedAccount] = useState<User | Org | null>(
+    null,
+  );
+
+  const [search, setSearch] = useState("");
+  const [open, setOpen] = useState(false);
+
+  const { users: minimalUsers } = useUsers();
+  const { orgs: minimalOrgs } = useOrgs();
+
+  const { orgs } = useOrgsWithLinks({
+    orgAccountGroupIds: minimalOrgs?.map((org) => org.accountGroupId),
+  });
+  const { users } = useUsersWithLinks({
+    userAccountIds: minimalUsers?.map((user) => user.accountId),
+  });
+
+  const options = useMemo(
+    () =>
+      [...(users ?? []), ...(orgs ?? [])].filter(
```

Comment:
You could take `minimalUsers/minimalOrgs` if `orgs`/`users` isn't yet loaded so that the data can load quicker, then avatars can appear when the full ones load... since this is fetching _all_ users and orgs with stuff linked to them, it's going to take a while to load when there is any amount of data in the real app

Reply:
Good idea, have addressed in https://github.com/hashintel/hash/pull/3392/commits/84788ad6de6361189b87f99fa3966b202f2eade3